packages/wallet/src/rollup-state-validator.ts

```diff
@@ -0,0 +1,285 @@
+/* External Imports */
+import * as AsyncLock from 'async-lock'
+
+import {
+  ChecksumAgnosticIdentityVerifier,
+  serializeObject,
+  SignatureVerifier,
+  DB,
+  SparseMerkleTree,
+  SparseMerkleTreeImpl,
+  BigNumber,
+  ONE,
+  runInDomain,
+  MerkleTreeInclusionProof,
+  ZERO,
+  getLogger,
+  bufToHexString,
+  hexStrToBuf,
+} from '@pigi/core'
+
+/* Internal Imports */
+import {
+  Address,
+  Balances,
+  Swap,
+  Transfer,
+  isSwapTransition,
+  isCreateAndTransferTransition,
+  isTransferTransition,
+  RollupTransaction,
+  SignedTransaction,
+  UNISWAP_ADDRESS,
+  UNI_TOKEN_TYPE,
+  PIGI_TOKEN_TYPE,
+  TokenType,
+  State,
+  StateUpdate,
+  StateSnapshot,
+  InclusionProof,
+  StateMachineCapacityError,
+  SignatureError,
+  AGGREGATOR_ADDRESS,
+  abiEncodeTransaction,
+  abiEncodeState,
+  parseStateFromABI,
+  DefaultRollupStateMachine,
+  InsufficientBalanceError,
+  NegativeAmountError,
+  InvalidTransactionTypeError,
+  InvalidTokenTypeError,
+  isStateTransitionError,
+  ValidationOutOfOrderError,
+  AggregatorUnsupportedError,
+  DefaultRollupBlock,
+} from './index'
+
+import {
+  RollupBlock,
+  RollupStateValidator,
+  RollupTransitionPosition,
+  FraudCheckResult,
+  RollupStateMachine,
+  RollupTransition,
+  SlippageError,
+  LocalMachineError,
+  LocalFraudProof,
+  UniTokenType,
+  PigiTokenType,
+} from './types'
+
+import { Transaction } from 'ethers/utils'
+import {
+  parseTransitionFromABI,
+  parseTransactionFromABI,
+} from './serialization'
+
+const log = getLogger('rollup-guard')
+export class DefaultRollupStateValidator implements RollupStateValidator {
+  public rollupMachine: DefaultRollupStateMachine
+  public currentPosition: RollupTransitionPosition = {
```

Comment:
Can make this private in favor of the getter on line 101. 

Note: that getter would let an external actor change the `currentPosition` from outside of this class, so exposing that is generally frowned upon unless it's absolutely required. Can return a copy (`return {...this.currentPosition}`) to get around that.

Reply:
Cool, I never knew why your code had that! 😂